### PR TITLE
Add handling for API Rate limit error

### PR DESCRIPTION
**Pull request by guyp-descope**
Add handling for API Rate limit error so it will throw a specific RateLimitException exception.

## Must

- [X] Tests
- [X] Documentation (if applicable)

### CODE REVIEW

File: descope/auth.py [hunk 1]
```diff
@@ -89,6 +92,19 @@ def do_get(
             verify=self.secure,
         )
         if not response.ok:
+            if response.status_code == 429:
```
Comment (dorsha) [inline descope/auth.py:95]: Let's extract to function and use it also below

**Review by dorsha — COMMENTED** [review]
Nice!

**Comment by dorsha** [comment]
Let's also add example to examples and README on how to handle the new error.

File: README.md [hunk 2]
```diff
@@ -615,6 +615,23 @@ updated_jwt = client.mgmt.jwt.updateJWT(
 )
 ```
 
+## API Rate limits
+
+Handle API rate limits by comparing the exception to the APIRateLimitExceeded exception, which includes the RateLimitParameters map with the key "Retry-After." This key indicates how many seconds until the next valid API call can take place. More information on Descope's rate limit is covered here: https://docs.descope.com/rate-limit
```
Comment (dorsha) [inline README.md:620]: For the link use markdown link option

**Review by dorsha — COMMENTED** [review]
Cool

File: README.md (hunk 2 above)
Comment (guyp-descope) [reply README.md]: Fixed

**Review by dorsha — APPROVED** [review]

**Pull request merged**